File: OCH_Tao/ABC/C/368.js
const H = require("fs").readFileSync("/dev/stdin", "utf8").trim().split("\n")[1].split(" ").map(x => parseInt(x, 10));
let T = 0;
for (const i of H) {
  T += Math.floor(i / 5) * 3;
  let M = i % 5;
  while (M > 0) {
    T++;
    if (T % 3 === 0) {
      M -= 3;
    } else {
      M--;
    }
  }
}
console.log(T);

File: OCH_Tao/ABC/C/368.py
N = int(input())
H = list(map(int,input().split()))
T = 0
for i in H:
  T+=i//5*3
  M = i%5
  while M>0:
    if T%3==2:
      T+=1
      M-=3
    else:
      T+=1
      M-=1
print(T)